extensions/hashicorp-vault/src/test/java/org/eclipse/dataspaceconnector/core/security/hashicorpvault/HashicorpCertificateResolverTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright (c) 2022 Mercedes-Benz Tech Innovation GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Mercedes-Benz Tech Innovation GmbH - Initial Test
+ *
+ */
+
+package org.eclipse.dataspaceconnector.core.security.hashicorpvault;
+
+import org.eclipse.dataspaceconnector.common.testfixtures.X509CertificateTestUtil;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+
+class HashicorpCertificateResolverTest {
+    private static final String KEY = "key";
+
+    // mocks
+    private HashicorpCertificateResolver certificateResolver;
+    private HashicorpVault vault;
+
+    @BeforeEach
+    void setup() {
+        vault = Mockito.mock(HashicorpVault.class);
+        final Monitor monitor = Mockito.mock(Monitor.class);
+        certificateResolver = new HashicorpCertificateResolver(vault, monitor);
+    }
+
+    @Test
+    void resolveCertificate() throws RuntimeException, IOException {
```

Comment:
That would mean we have no unit tests for the certificate resolver at all. Is that okay?

Reply:
@bcronin90 in my opinion you could choose to test this in two ways:
- with an Integration Test, that will cover the client part to resolve a secret and parsing it to a certificate
- with an unit test, that mocks the vault response and also mocks a "pem parser" component that transforms a string in a certificate (and that's tested separately), and verifies eventual edge cases.

I would keep either one or the other, not both (probably I would go with the latter, since the Vault client it's already integration-tested by the `HashicorpVaultIntegrationTest`)